image/README.md

```diff
@@ -1,113 +1,193 @@
-4. Ensure read and write access to `/dev/kvm` (and repeat after every reboot)
+- Install tools:
+
+    <details>
+    <summary>Ubuntu / Debian</summary>
+
+    ```sh
+    sudo apt-get update
+    sudo apt-get install --assume-yes --no-install-recommends \
+        dnf \
+        systemd-container \
+        qemu-system-x86 \
+        qemu-utils \
+        ovmf \
+        e2fsprogs \
+        squashfs-tools \
+        efitools \
+        sbsigntool \
+        coreutils \
+        curl \
+        jq \
+        util-linux \
+        virt-manager
+    ```
 
-    ```shell-session
-    sudo chmod 666 /dev/kvm
+    </details>
+
+    <details>
+    <summary>Fedora</summary>
+
+    ```sh
+    sudo dnf install -y \
+        edk2-ovmf \
+        systemd-container \
+        qemu \
+        e2fsprogs \
+        squashfs-tools \
+        efitools \
+        sbsigntools \
+        coreutils \
+        curl \
+        jq \
+        util-linux \
+        virt-manager
     ```
 
-## Configuration
+    </details>
 
-Create a configuration file in `image/config.mk` to override any of the variables found at the top of the [Makefile](Makefile).
-Important settings are:
+- Prepare secure boot PKI (see `secure-boot/genkeys.sh`)
 
-- `BOOTSTRAPPER_BINARY`: path to a bootstrapper binary. Can be substituted with a path to a `debugd` binary if a debug image should be built. The binary has to be built before!
-- `CONTAINER_ENGINE`: container engine used to run COSA. either `podman` or `docker`.
-- `COSA_INIT_REPO`: Git repository containing CoreOS config. Cloned in `cosa-init` target.
-- `COSA_INIT_BRANCH`: Git branch checked out from `COSA_INIT_REPO`. Can be used to test out changes on another branch before merging.
-- `NETRC` path to a netrc file containing a GitHub PAT. Used to authenticate to GitHub from within the COSA container.
-- `GCP_IMAGE_NAME`: Image name for the GCP image. Set to include a timestamp when using the build pipeline. Can be set to a custom value if you want to upload a custom image for testing on GCP.
-- `AZURE_IMAGE_NAME`: Image name for the Azure image. Can be set to a custom value if you want to upload a custom image for testing on Azure.
+## Build
 
-Example `config.mk` to create a debug image with docker and name it `my-custom-image`:
+When building your first image, prepare the secure boot PKI (see `secure-boot/genkeys.sh`) for self-signed, locally built images.
 
-```Makefile
-BOOTSTRAPPER_BINARY = ../build/debugd
-CONTAINER_ENGINE = docker
-GCP_IMAGE_NAME = my-custom-image
-AZURE_IMAGE_NAME = my-custom-image
-```
+After that, you can build the image with:
 
-## Build an image
+```sh
+# OPTIONAL: to create a debug image, export the following line
+# export BOOTSTRAPPER_BINARY=$(realpath ${PWD}/../../build/debugd)
+# OPTIONAL: specify custom path to secure boot PKI
+# export PKI=/path/to/pki/folder
+sudo make -j $(nproc)
```

Comment:
This option seems to not be respected by all of the mkosi workflow:

```shell
$ sudo PKI=./pki_testing/ make -j $(nproc)                                                                                                                                                                                                               
mkdir -p /home/daniel/test/constellation/image/mkosi.extra/usr/bin                                                                                                                                                                                        
# for auto enrollment using systemd-boot (not working yet)                                                                                                                                                                                                    
mkdir -p "/home/daniel/test/constellation/image/mkosi.extra/boot/loader/keys/auto"                                                                                                                                                                        
mkdir -p /home/daniel/test/constellation/image/mkosi.extra/usr/sbin
[...]
Can't load key from file '/home/daniel/test/constellation/image/pki/db.key'
809BB8AC1B7F0000:error:80000002:system library:BIO_new_file:No such file or directory:crypto/bio/bss_file.c:67:calling fopen(/home/daniel/test/constellation/image/pki/db.key, r)
809BB8AC1B7F0000:error:10000080:BIO routines:BIO_new_file:no such file:crypto/bio/bss_file.c:75:
```

Reply:
Thanks. I think we will need to symlink the used pki to `./pki` for mkosi to work. Will update the docs.